pkg/templates/stamper.go

```diff
@@ -195,52 +193,49 @@ func (s *Stamper) applyTemplate(resourceTemplateJSON []byte) (*unstructured.Unst
-	ctx, cancel := context.WithTimeout(ctx, 1*time.Second)
-	defer cancel()
-
-	ytt := "ytt"
-	// ko copies the content of the kodata directory into the container at a path referenced by $KO_DATA_PATH
-	if kodata, ok := os.LookupEnv("KO_DATA_PATH"); ok {
-		ytt = path.Join(kodata, fmt.Sprintf("ytt-%s-%s", runtime.GOOS, runtime.GOARCH))
-	}
-
-	args := []string{"-f", "-"}
-	stdin := bytes.NewReader([]byte(template))
-	stdout := bytes.NewBuffer([]byte{})
-	stderr := bytes.NewBuffer([]byte{})
+	yttOpts := yttcmd.NewOptions()
 
 	// inject each key of the template context as a ytt value
 	templateContext := map[string]interface{}{}
 	b, err := json.Marshal(s.TemplatingContext)
 	if err != nil {
-		// NOTE we can ignore subsequent json errors, if there's a issue with the data it will be caught here
+		// NOTE we can ignore subsequent json errors, if there's an issue with the data it will be caught here
 		return nil, fmt.Errorf("unable to marshal template context: %w", err)
 	}
-	_ = json.Unmarshal(b, &templateContext)
+	err = json.Unmarshal(b, &templateContext)
+	if err != nil {
+		// NOTE this should never error because we have already checked the error after json.Marshal
+		return nil, err
+	}
+
+	var dataValues []string
 	for k := range templateContext {
 		raw, _ := json.Marshal(templateContext[k])
-		args = append(args, "--data-value-yaml", fmt.Sprintf("%s=%s", k, raw))
+		dataValues = append(dataValues, fmt.Sprintf("%s=%s", k, raw))
 	}
 
-	cmd := exec.CommandContext(ctx, ytt, args...)
-	cmd.Stdin = stdin
-	cmd.Stdout = stdout
-	cmd.Stderr = stderr
+	// equivalent to `--data-value-yaml`
+	yttOpts.DataValuesFlags.KVsFromYAML = dataValues
 
-	log.V(logger.DEBUG).Info("ytt call", "args", args, "input", template)
-	if err := cmd.Run(); err != nil {
-		msg := stderr.String()
-		if msg == "" {
-			return nil, fmt.Errorf("unable to apply ytt template: %w", err)
-		}
-		return nil, fmt.Errorf("unable to apply ytt template: %s", msg)
+	input := yttcmd.Input{
+		Files: []*yttfiles.File{yttfiles.MustNewFileFromSource(yttfiles.NewBytesSource("stdin.yml", []byte(template)))},
+	}
+	noopUI := yttui.NewCustomWriterTTY(false, bytes.NewBuffer([]byte{}), bytes.NewBuffer([]byte{}))
+
+	log.V(logger.DEBUG).Info("ytt call", "data values", dataValues, "input", template)
+	output := yttOpts.RunWithFiles(input, noopUI)
+	if output.Err != nil {
+		return nil, fmt.Errorf("unable to apply ytt template: %w", output.Err)
+	}
+
+	outputBytes, err := output.DocSet.AsBytes()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get output as bytes: %w", err)
 	}
-	output := stdout.String()
-	log.V(logger.DEBUG).Info("ytt result", "output", output)
+	log.V(logger.DEBUG).Info("ytt result", "output", string(outputBytes))
 
 	stampedObject := &unstructured.Unstructured{}
-	if err := yaml.Unmarshal([]byte(output), stampedObject); err != nil {
+	if err = yaml.Unmarshal(outputBytes, stampedObject); err != nil {
```

Comment:
nit

```suggestion
	if err := yaml.Unmarshal(outputBytes, stampedObject); err != nil {
```

Reply:
curious- why do you want this change? `err` is already declared above